Add documentation for mobile: source endpoint

mykola-mokhnach opened the pull request:
Based on https://github.com/appium/appium-xcuitest-driver/pull/534

Commit:
Add documentation for mobile: source endpoint

imurchie commented on docs/en/writing-running-appium/ios-xctest-source.md:
```diff
@@ -0,0 +1,22 @@
+## Getting Page Source For iOS With WebDriverAgent/XCTest Backend
+
+WebDriverAgent provides a possibility to represent page source in different formats. Usually it is enough to call _getPageSource_ driver method, which is the standard one and is available for all the client libraries. This method returns the hierarchical representation of the current UI source represented as XML document. However, sometimes it is convenient to get page representation in JSON form ...
```
This makes it seem like "xml", "json" and some other format(s) are available, but the `format` description below says only the two are there.

mykola-mokhnach replied:
yes, I want to add at least one more and there is a PR for that, however I don't know when it is going to be merged

imurchie reviewed (changes requested):
I think that throughout what is in italics should be in code format (with backticks).

jlipps reviewed (commented):
I like this but this should be added as a new command doc, using the command template and doc generator that @dpgraham has made. We should in fact have a whole section for "mobile" commands, and this should be one of them. @dpgraham are you up for working on this branch with @mykola-mokhnach and helping to put this in that format?

dpgraham commented:
Yeah, I can easily add a section to the commands documentation. I'll just cut-and-paste what's in this PR.

Commit:
Add one more supported source format

mykola-mokhnach commented:
@imurchie What else should be done to have this document merged?

imurchie commented on docs/en/writing-running-appium/ios-xctest-source.md:
```diff
@@ -0,0 +1,22 @@
+## Getting Page Source For iOS With WebDriverAgent/XCTest Backend
+
+WebDriverAgent provides a possibility to represent page source in different formats. Usually it is enough to call _getPageSource_ driver method, which is the standard one and is available for all the client libraries. This method returns the hierarchical representation of the current UI source represented as XML document. However, sometimes it is convenient to get page representation in JSON form ...
```
The `source` command has a page: `/docs/en/commands/session/source.md`

imurchie reviewed (commented):
A couple of things:
1. I still would like all the things in _italics_ to be in `backticks`.
2. This should be in the ios folder (https://github.com/appium/appium/tree/master/docs/en/writing-running-appium/ios)
3. This should be linked to in the 'source' document (https://github.com/appium/appium/blob/master/commands-yml/commands/session/source.yml)

mykola-mokhnach commented:
> I still would like all the things in italics to be in backticks.

Changed 

> This should be in the ios folder (https://github.com/appium/appium/tree/master/docs/en/writing-running-appium/ios)

Moved

> This should be linked to in the 'source' document (https://github.com/appium/appium/blob/master/commands-yml/commands/session/source.yml)

I'm not sure into which category this should be added. Any hints there?

Commit:
Address review comments

imurchie commented:
I would put it in the description,  in the source documentation.

mykola-mokhnach commented:
NP, I can do that when the new documentation is published

imurchie commented:
Just a link is all that is needed, I think. Something like "For iOS you can get a different underlying source structure by using [link]"
